eth/beacon/state_machines/base.py

```diff
@@ -83,25 +83,30 @@ def get_block_class(cls) -> Type[BaseBeaconBlock]:
             return cls.block_class
 
     @classmethod
-    @functools.lru_cache(maxsize=32)
     @to_tuple
-    def get_prev_blocks_to_slot(cls,
-                                last_block_hash: Hash32,
-                                chaindb: BaseBeaconChainDB,
-                                depth: int,
-                                start_slot: int) -> Iterable[BaseBeaconBlock]:
+    def get_prev_blocks(cls,
+                        last_block_hash: Hash32,
+                        chaindb: BaseBeaconChainDB,
+                        searching_depth: int,
+                        target_slot_number: int) -> Iterable[BaseBeaconBlock]:
```

Comment:
Maybe `min_slot_number` since minimum is somewhat of an *optional* constraint and `target` is likely to be interpreted as a firm constraint.  With the term target, my intuition would say that this search would continue until it hit this slot number.